apps/events/tests/model_tests.py

```diff
@@ -304,6 +306,53 @@ def test_user_group_rule(self):
         self.assertTrue(response["status"])
         self.assertEqual(212, response["status_code"])
 
+    def test_rule_offset_and_mark_offset(self):
+        group: Group = G(Group, name="Testgroup")
+        group_rule: GradeRule = G(UserGroupRule, group=group, offset=24)
+        rule_bundle: RuleBundle = G(RuleBundle, description="")
+        rule_bundle.user_group_rules.add(group_rule)
+        self.attendance_event.rule_bundles.add(rule_bundle)
+        # Registration just opened
+        self.attendance_event.registration_start = timezone.now() - timezone.timedelta(
+            minutes=1
+        )
+
+        # User should not be able to register because of the group rule offset
+        self.user.groups.add(group)
+        response = self.attendance_event.is_eligible_for_signup(self.user)
+        self.assertFalse(response["status"])
+        self.assertEqual(422, response["status_code"])
+
+        # Simulate a day passing for the offset to not have an effect anymore
+        self.attendance_event.registration_start -= timezone.timedelta(hours=24)
+        self.attendance_event.save()
+
+        # User should be able to register for event since offset has passed
+        response = self.attendance_event.is_eligible_for_signup(self.user)
+        self.assertTrue(response["status"])
+        self.assertEqual(212, response["status_code"])
```

Comment:
I would prefer if the codes are linked to the relevant class-values, since that would make it clearer what kind of value we are looking after here

Reply:
I agree that it could be nice, and I almost went for implementing it. Would be a lot more descriptive for users.

I ended up not doing it because it could end up being a lot of work, and not actually needed to correct the problem at hand.

Reply:
I would love if you took the time to add it! We think we should prioritize making it as clear as possible to new developers (which often includes ourselves)

Reply:
I have tried, but I can say by half way that it is a nightmare :upside_down_face:

Reply:
But in this case isn't it just to replace it with`TypeOfRule.some_type_of_code`? Or is there something I'm missing?

Reply:
While that works OK it does little for the actual implementation of the rule bundles. You'll get some clarity on the status code meanings themselves, but we end up with a duplication of their actual meanings inside of the _rule satisfaction_ methods anyways. All of those response dictionaries which are passed around and mutated all the time are the things that actually make this confusing and hard to work with. Just adding an Enum with the codes just adds to that without taking away any of the complexity

Reply:
I think this test is fine, just that I would love to not have to have a tab of the status-codes open at the same time to check that the codes are the correct ones 😅 
I'm not sure what you are talking about about mutating dictionaries? This test just reasignes `response` every time you do a new check?

Reply:
You can have a look at #2480 for changes of the signup thingies if you want to